central/auth/datastore/singleton.go

```diff
@@ -15,7 +24,26 @@ var (
 // Singleton provides a singleton auth machine to machine DataStore.
 func Singleton() DataStore {
 	once.Do(func() {
-		ds = New(pgStore.New(globaldb.GetPostgres()))
+		set := m2m.TokenExchangerSetSingleton(roleDataStore.Singleton(), jwt.IssuerFactorySingleton())
+		ds = New(pgStore.New(globaldb.GetPostgres()), set)
+
+		// On initialization of the store, list all existing configs and fill the set.
+		ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS), sac.ResourceScopeKeys(resources.Access)))
```

Comment:
Nit: `ctx` is needed only in `createTokenExchangersForExistingConfigs`, please move it there.